airflow/www/extensions/init_views.py

```diff
@@ -308,5 +318,16 @@ def init_api_experimental(app):
         "The authenticated user has full access.",
         RemovedInAirflow3Warning,
     )
+    base_paths.append("/api/experimental")
     app.register_blueprint(endpoints.api_experimental, url_prefix="/api/experimental")
     app.extensions["csrf"].exempt(endpoints.api_experimental)
+
+
+def init_api_auth_provider(app):
+    """Initialize the API offered by the auth manager."""
+    auth_mgr = get_auth_manager()
+    blueprint = auth_mgr.get_api_blueprint()
+    if blueprint is not None:
+        base_paths.append(blueprint.url_prefix)
+        app.register_blueprint(blueprint)
+        app.extensions["csrf"].exempt(blueprint)
```

Comment:
Hmm, this doesn't seem right to me. Surely we want csrf protection?

Reply:
Happy to remove it but this is what we do with all Rest APIs in Airflow. See https://github.com/apache/airflow/blob/main/airflow/www/extensions/init_views.py#L275

Reply:
We can't assume auth managers will only add api endpoints though?

Reply:
What do you mean? What else do you expect?

Reply:
I might have understood. We assume (and maybe we should force?) that `get_api_blueprint` is only used to expose new Rest API endpoints

Reply:
As an example, FAB auth manager has the ui for managing roles/users. Other auth managers technically could do something similar. Blueprint != api ultimately, right? A better example to follow would be how [plugins register - no csrf exemption](https://github.com/apache/airflow/blob/25cd12d307be6359c844f4764b6f90e954bccb56/airflow/www/extensions/init_views.py#L158).

Reply:
Correct but here on line `333`: `app.extensions["csrf"].exempt(blueprint)`, we only add the exception to the blueprint right? And the blueprint is only Rest API related

Reply:
Ah, I didn't notice the name. Does restricting it to api really make sense though? I wonder if it should be [opt-out like this](https://airflow.apache.org/docs/apache-airflow/stable/authoring-and-scheduling/plugins.html#exclude-views-from-csrf-protection), and allow auth manager to just register any old blueprint.

Reply:
I'm torn on this. We can't really "force" users to only specify apis, best we can do is document that csrf protection is off.

cc @potiuk @pierrejeambrun @hussein-awala

Reply:
Why not? Why cannot we force users to only specify apis?

Reply:
Sorry, I missed the typing change there. Feels weird to force the use of connexion.FlaskApi on auth managers no? But yes, we could.

But high level, I'm still torn on whether we should it should be "anything" or "only api".

Reply:
We dont force. In `base_auth_manager` it returns `None` by default, so only if you need, you can add Rest API endpoints through the auth manager. That's an optional step when you implement your auth manager

Reply:
Yes of course, all of this is optional. I mean "force" in the sense that if an author wanted an http endpoint, they'd have to use connexion.FlaskApi instead of a more common flask.Blueprint.

Reply:
I am fine with letting the user use common `flask.Blueprint`, the only thing is we need to trust/assume the user will use it only for Rest API purposes. As explained in this [comment](https://github.com/apache/airflow/pull/34349#discussion_r1349194668), in Airflow we make a difference between a Rest API and a view

Reply:
Should we have TWO methods instead ? One for APIs and one for UI ? 

```python
def get_api_endpoints():
   # return API endpoints

def get_ui_blueprints():
   # return UI blueprints
```

The first would be without CSRF the seond with. I think that would make more sense. They could (should ?) be even intitialized in different place.

Reply:
I like it... simple solution.

Reply:
Sounds good to me. Do we want to return `flask.Blueprint` or `FlaskApi` for `get_api_endpoints`? Also, since for now we do not need `get_ui_blueprints`, do you still want to have it? We could also add it when needed

Reply:
> Sounds good to me. Do we want to return flask.Blueprint or FlaskApi for get_api_endpoints? Also, since for now we do not need get_ui_blueprints, do you still want to have it? We could also add it when needed

* I think FlaskAPI is better

* I believe we need to add get_ui_blueprints - should not we have the whole "Security" menu added via blueprints ? This whole menu section should be gone where FAB auth manager is not configured.

I think this could be a separate PR, but it's still needed.

Reply:
This is currently done in the FAB security manager override. See: https://github.com/apache/airflow/blob/main/airflow/auth/managers/fab/security_manager/override.py#L153. I just moved the implementation from the security manager to the override. Blueprints were never used for that. Do you want to replace this implementation and use blueprints for that purpose?

Reply:
Ah no . Then it's perfectly fine to leave it there - even better this way I think

Reply:
I renamed `get_api_blueprint` by `get_api_endpoints` and modify the return type (`FlaskApi` instead of `Blueprint`). `get_ui_blueprints` can be implemented later when needed.

Reply:
@jedcunningham